pkg/ingester/client/circuitbreaker.go

```diff
@@ -69,7 +69,13 @@ func NewCircuitBreaker(inst ring.InstanceDesc, cfg CircuitBreakerConfig, metrics
 			transitionHalfOpen.Inc()
 			level.Info(logger).Log("msg", "circuit breaker is half-open", "ingester", inst.Id, "previous", event.OldState, "current", event.NewState)
 		}).
-		HandleIf(func(r any, err error) bool { return isFailure(err) }).
+		HandleIf(func(r any, err error) bool {
```

Comment:
This method isn't the correct way to log an individual failure. Instead, the log should be in the various `OnX` methods above.